kimchi/src/proof.rs

```diff
@@ -144,6 +157,12 @@ impl<G: AffineCurve> RecursionChallenge<G> {
     }
 }
 
+impl<G: AffineCurve> ConsecutiveEvals<G> {
+    pub fn array(&self) -> [&ProofEvaluations<Vec<ScalarField<G>>>; 2] {
```

Comment:
I don't necessarily like these functions because they are error prone (what if we add an evaluation at some point and forget to update this function?)

ideas: deriving this with a macro, or using something [like this](https://stackoverflow.com/questions/54177438/how-to-programmatically-get-the-number-of-fields-of-a-struct) as an easier solution to make sure we created an array of the right size, or just accepting this tiny risk

Reply:
I see the point, if we do it, it should better be a constant. The reason that I convert to an array is because in the prover we tend to iterate over the two elements of `ConsecutiveEvals` (meaning the zeta and the zeta_omega), together with other tuples in the code (for example, using them in `zip` calls). Other solutions that come to mind seem more verbose 🤔